src/main/webapp/app/shared/delete-dialog/delete-button.directive.ts

```diff
@@ -1,14 +1,19 @@
-import { DeleteDialogData, DeleteDialogService } from 'app/shared/delete-dialog/delete-dialog.service';
-import { Output, EventEmitter, Input, Directive, HostListener, Renderer2, ElementRef, OnInit } from '@angular/core';
+import { DeleteDialogService } from 'app/shared/delete-dialog/delete-dialog.service';
+import { Input, Directive, HostListener, Renderer2, ElementRef, OnInit, EventEmitter, Output } from '@angular/core';
 import { TranslateService } from '@ngx-translate/core';
+import { ActionType, DeleteDialogData } from 'app/shared/delete-dialog/delete-dialog.model';
 
 @Directive({ selector: '[jhiDeleteButton]' })
 export class DeleteButtonDirective implements OnInit {
     @Input() entityTitle: string;
     @Input() deleteQuestion: string;
     @Input() deleteConfirmationText: string;
     @Input() additionalChecks?: { [key: string]: string };
+    @Input() actionType: ActionType = ActionType.Delete;
     @Output() delete = new EventEmitter<{ [key: string]: boolean }>();
+    @Input() set closeDialogTrigger(value: boolean) {
```

Comment:
Why aren't we using some kind of observable here?

Reply:
I like that idea, but in my opinion it would be an overkill for each component to create an observable, that will only emit once. Passing `boolean` value is a lot simpler in terms of reusability.